### PR TITLE
Added lint report for fe2

### PR DESCRIPTION
**Pull request by matteosz**
After a conversation about Timber-lint in the PR #1710 , here's an integration on the FE2 CI that generates a lint report and trigger a static code analysis based on it

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -227,6 +227,19 @@ jobs:
...
 
+      - name: Generate Sarif Lint Report
+        shell: bash
+        run: |
+          # To run the CI with debug informations, add --info
+          ./gradlew lintDebug --parallel --build-cache
+      
+      - name: Upload lint sarif
+        uses: github/codeql-action/upload-sarif@v2
+        if: success() || failure()
```
Comment (pierluca) [inline .github/workflows/ci.yaml:238]: why this `if` clause here but neither before nor after ?

Comment (matteosz) [reply .github/workflows/ci.yaml:238]: No particular reason as the order doesn't change the workflow

Comment (jbsv) [reply .github/workflows/ci.yaml:238]: instead of `success() || failure()` why not use `always()`?

**Review by jbsv — CHANGES_REQUESTED** [review]
maybe a way to KIS

**Review by pierluca — APPROVED** [review]

**Pull request merged**